### PR TITLE
Check all feeds before aborting

### PR DESCRIPTION
**Pull request by bmos**
Queue until end and abort if any error was raised

### CODE REVIEW

**Comment by bmos** [comment]
going to add Brett's suggestions
https://discussion.dsausa.org/t/dsa-feed-not-updating-github-actions/31234/15?u=wil.t

**Comment by bmos** [comment]
This is a big improvement, but the one thing that still stops it in its tracks are ConnectionErrors from Faraday.
For some reason, some URLs (Houston DSA at least) are causing a timeout trying to open the connection which triggers this ConnectionError. Because it doesn't get an HTML status code, it doesn't continue with the other tests.
It's also the same situation with SSL errors which raise a ConnectionError.

**Comment by bmos** [comment]
Feel free to rewrite any of this, as mentioned I have never touched Ruby before.

File: tests/feedcheck.rb [hunk 1]
```diff
@@ -26,41 +33,48 @@
...
   end
-  print '✓ '
   # Check is the XML actually parses as XML
-  xml = Faraday.get(URI(feed)).body
+  xml = faraday.get(URI(feed)).body
   xml_err = Nokogiri::XML(xml).errors
-  unless xml_err.empty?
-    print "✗\nUnusable XML syntax: #{feed}\n#{xml_err}"
-    abort
+    unless xml_err.empty?
```
Comment (brettchalupa) [inline tests/feedcheck.rb:61]: nit: in Ruby this line would be unindented

**Review by brettchalupa — COMMENTED** [review]
Great improvements! Looking good! This will make it easy to fix the busted feeds

**Comment by bmos** [comment]
I believe this to be done, unless someone wants to make a pass at logging and ignoring SSL and connection issues.

**Comment by brettchalupa** [comment]
@bmos for the SSL errors, I think just commenting out those is the best option. I checked out a few and they have expired SSL certs, which IMO means we shouldn't use them.

**Comment by bmos** [comment]
Certainly, but we want the action log to continue checking the other sites rather than getting 'stuck' like it did for redirects prior to this PR

**Review by brettchalupa — APPROVED** [review]
Working really well for me locally! I don't have merge power yet, but I think this will make it a lot easier to maintain into the future.

**Comment by brettchalupa** [comment]
@bmos yes! Breaking the array of feeds up into threads could help. Here's a generic example:

``` ruby
# Sample array
arr = [1, 2, 3, 4, 5]

# Define the operation to perform on each element
def process_element(element)
  sleep 1 # Simulate IO-bound operation
  element * 2
end

# Create threads and collect results
results = []
threads = arr.map do |element|
  Thread.new do
    results << process_element(element)
  end
end

# Wait for all threads to finish
threads.each(&:join)

# Print the results
puts results # => [2, 4, 6, 8, 10] (order might vary)
```

It should, in theory, be more efficient. It might require some slicing up the arrays into multiple. Let me know if I can help adapt given the example above is maybe too generic. 😂

**Comment by bmos** [comment]
wow that was so much easier than implementing threading in any other language I have tried.

**Comment by bmos** [comment]
It now runs all the way through the whole file without getting stuck.

**Pull request merged**